### PR TITLE
add job for running codecov metrics script

### PR DESCRIPTION
**Pull request by estute**
Job for running https://github.com/edx/testeng-ci/pull/158

### CODE REVIEW

File: testeng/jobs/gatherCodecovMetrics.groovy [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    description('Gather metrics on how reliably codecov is posting statuses on pull requests')
+
+    authorization GENERAL_PRIVATE_JOB_SECURITY()
+
+    parameters {
+        stringParam('PULL_REQUEST_TIME_FRAME', '3600',
+                    'blah')
+        stringParam('CODECOV_TIME_FRAME', '1800',
+                    'blah')
```
Comment (jmbowman) [inline testeng/jobs/gatherCodecovMetrics.groovy]: Shouldn't "blah" get replaced with actual descriptions for these?

File: testeng/jobs/gatherCodecovMetrics.groovy [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        }
+    }
+
+    steps {
+        shell(readFileFromWorkspace('testeng/resources/gather-codecov-metrics.sh'))
+    }
+
+    publishers {
+        mailer('testeng@edx.org')
+        hipChatNotifier JENKINS_PUBLIC_HIPCHAT('')
```
Comment (jmbowman) [inline testeng/jobs/gatherCodecovMetrics.groovy]: What is the argument here supposed to represent?

Comment (estute) [reply testeng/jobs/gatherCodecovMetrics.groovy]: from the docs 'when left empty, the token specified in the global configuration will be used.'.

File: testeng/jobs/gatherCodecovMetrics.groovy (hunk 1 above)
Comment (estute) [reply testeng/jobs/gatherCodecovMetrics.groovy]: thanks. that was a placeholder until i though of something better

**Pull request merged**